File: src/main/java/io/github/tanguygab/playerlistexpansion/filters/Placeholder.java
package io.github.tanguygab.playerlistexpansion.filters;

import me.clip.placeholderapi.PlaceholderAPI;
import org.bukkit.OfflinePlayer;

import java.util.ArrayList;
import java.util.List;

public class Placeholder extends Filter {

    private final List<String[]> placeholders = new ArrayList<>();

    public Placeholder(String arg) {
        for (String str : split(arg))
            placeholders.add(str.split("="));
    }

    @Override
    public boolean filter(String name, OfflinePlayer viewer) {
        for (String[] arr : placeholders)
            if (parse(name,viewer,arr))
                return true;
        return false;
    }

    private boolean parse(String name, OfflinePlayer viewer, String[] arr) {
        return PlaceholderAPI.setPlaceholders(getOffline(name), arr[0]).equals(PlaceholderAPI.setPlaceholders(viewer, arr[1]));
    }
}

File: src/main/java/io/github/tanguygab/playerlistexpansion/filters/Placeholder.kt
package io.github.tanguygab.playerlistexpansion.filters

import me.clip.placeholderapi.PlaceholderAPI
import org.bukkit.OfflinePlayer

class Placeholder(arg: String) : Filter() {
    private val placeholders = split(arg).map { it.split("=") }

    override fun filter(name: String?, viewer: OfflinePlayer?): Boolean {
        return placeholders.any { parse(name, viewer, it) }
    }

    private fun parse(name: String?, viewer: OfflinePlayer?, arr: List<String>): Boolean {
        return PlaceholderAPI.setPlaceholders(getOffline(name), arr[0]) ==
                PlaceholderAPI.setPlaceholders(viewer, arr[1])
    }
}
